api/v1alpha1/ndb_api_helpers.go

```diff
@@ -155,43 +156,123 @@ func GetNoneTimeMachineSLA(ctx context.Context, ndbclient *ndbclient.NDBClient)
 	return sla, fmt.Errorf("NONE TimeMachine not found")
 }
 
-// Fetches all the profiles and returns a map of OOB profiles
-// Returns an error if one or more OOB profiles is not found
-func GetOOBProfiles(ctx context.Context, ndbclient *ndbclient.NDBClient, dbType string) (profileMap map[string]ProfileResponse, err error) {
-	// Map of profile type to profiles
-	profileMap = make(map[string]ProfileResponse)
+// +kubebuilder:object:generate:=false
+type ProfileResolver interface {
+	Resolve(ctx context.Context, allProfiles []ProfileResponse, filter func(p ProfileResponse) bool) (profile ProfileResponse, err error)
```

Comment:
Hey @yash-ntnx , allProfiles here refer to the response of GET /profiles.

Reply:
Hey @yash-ntnx @manavrajvanshi This may need some refactoring as these are the existing constructs. I am ok to take up the suggested change as a part of a follow-up PR after offline discussion. I will look into other open source operators to understand what is the industry-standard on this. Resolving this for now.